pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -544,26 +544,37 @@ func (ku *Upgrader) upgradeAgentScaleSets(ctx context.Context) error {
 
 				ku.logger.Infof("Copying custom annotations, labels, taints from old node %s to new node %s...", vmToUpgrade.Name, newNodeName)
 				ch := make(chan struct{}, 1)
-				go func() {
+
+				ctx, cancel := context.WithTimeout(context.Background(), nodePropertiesCopyTimeout)
+				defer cancel()
+
+				go func(ctx context.Context) {
 					for {
-						err = ku.copyCustomPropertiesToNewNode(client, strings.ToLower(vmToUpgrade.Name), strings.ToLower(newNodeName))
-						if err != nil {
-							ku.logger.Warningf("Failed to copy custom annotations, labels, taints from old node %s to new node %s: %v", vmToUpgrade.Name, newNodeName, err)
-							time.Sleep(time.Second * 5)
-						} else {
+						select {
+						case <-ctx.Done():
+							ku.logger.Errorf("Copying custom annotations, labels, taints from old node %s to new node %s can't complete within %v", vmToUpgrade.Name, newNodeName, nodePropertiesCopyTimeout)
 							ch <- struct{}{}
+							return
+						default:
+							err = ku.copyCustomPropertiesToNewNode(client, vmToUpgrade.Name, newNodeName)
+							if err != nil {
+								ku.logger.Warningf("Failed to copy custom annotations, labels, taints from old node %s to new node %s: %v", vmToUpgrade.Name, newNodeName, err)
+							} else {
+								ku.logger.Infof("Successfully copied custom annotations, labels, taints from old node %s to new node %s.", vmToUpgrade.Name, newNodeName)
+								ch <- struct{}{}
+								return
+							}
 						}
+
+						time.Sleep(time.Second * 5)
 					}
-				}()
-
-				for {
-					select {
-					case <-ch:
-						ku.logger.Infof("Successfully copied custom annotations, labels, taints from old node %s to new node %s.", vmToUpgrade.Name, newNodeName)
-					case <-time.After(nodePropertiesCopyTimeout):
-						ku.logger.Errorf("Copying custom annotations, labels, taints from old node %s to new node %s can't complete within %v", vmToUpgrade.Name, newNodeName, nodePropertiesCopyTimeout)
-					}
-					break
+				}(ctx)
+
+				select {
+				case <-ch:
+					ku.logger.Infof("Finished trying to copy custom annotations, labels, taints from old node %s to new node %s.", vmToUpgrade.Name, newNodeName)
+				case <-time.After(nodePropertiesCopyTimeout):
```

Comment:
I know this usage was carried over from the existing implementation, but I think this `time.After` calculation is unnecessary.

I think a more idiomatic implementation would be having two channels: one for success condition and one for error condition.

See here for a simple example:

https://github.com/Azure/aks-engine/blob/master/test/e2e/kubernetes/pod/pod.go#L289

Reply:
@jackfrancis , per your request, removed the time.After for the channel.